tensorflow/core/kernels/broadcast_to_op.cc

```diff
@@ -0,0 +1,239 @@
+            output = input.reshape(reshape).broadcast(broadcast);
+          } break;
+          case 3: {
+            auto input = input_tensor.tensor<T, 3>();
+            output = input.reshape(reshape).broadcast(broadcast);
+          } break;
+          case 4: {
+            auto input = input_tensor.tensor<T, 4>();
+            output = input.broadcast(broadcast);
+          } break;
+          default:
+            ctx->CtxFailure(errors::InvalidArgument(
+                "invalid shape to broadcast from ", input_shape.DebugString(),
+                " to ", output_shape.DebugString()));
+            break;
+        }
+      } break;
+      case 5: {
+        auto reshape = AsEigenDSizesWithPrefix<5>(input_shape);
+        auto broadcast = output_shape.AsEigenDSizes<5>();
+
+        BROADCAST_SHAPE(broadcast, reshape, 5, input_shape, output_shape);
+
+        auto output = output_tensor->tensor<T, 5>();
+        switch (input_shape.dims()) {
+          case 0: {
+            output.setConstant(input_tensor.scalar<T>()());
+          } break;
+          case 1: {
+            auto input = input_tensor.tensor<T, 1>();
+            output = input.reshape(reshape).broadcast(broadcast);
+          } break;
+          case 2: {
+            auto input = input_tensor.tensor<T, 2>();
+            output = input.reshape(reshape).broadcast(broadcast);
+          } break;
+          case 3: {
+            auto input = input_tensor.tensor<T, 3>();
+            output = input.reshape(reshape).broadcast(broadcast);
+          } break;
+          case 4: {
+            auto input = input_tensor.tensor<T, 4>();
+            output = input.reshape(reshape).broadcast(broadcast);
+          } break;
+          case 5: {
+            auto input = input_tensor.tensor<T, 5>();
+            output = input.broadcast(broadcast);
+          } break;
+          default:
+            ctx->CtxFailure(errors::InvalidArgument(
+                "invalid shape to broadcast from ", input_shape.DebugString(),
+                " to ", output_shape.DebugString()));
+            break;
+        }
+      } break;
+      default:
+        ctx->CtxFailure(errors::InvalidArgument(
+            "invalid shape to broadcast from ", input_shape.DebugString(),
+            " to ", output_shape.DebugString()));
+        break;
+    }
+  }
+
+ private:
+  template <int NDIMS>
+  Eigen::DSizes<Eigen::DenseIndex, NDIMS> AsEigenDSizesWithPrefix(
+      const TensorShape& shape) const {
+    Eigen::DSizes<Eigen::DenseIndex, NDIMS> dsizes;
+    for (int d = 0; d < NDIMS - shape.dims(); d++) {
+      dsizes[d] = 1;
+    }
+    for (int d = NDIMS - shape.dims(); d < NDIMS; d++) {
+      dsizes[d] = shape.dim_size(d - (NDIMS - shape.dims()));
+    }
+    return dsizes;
+  }
+};
+
+// As MakeShape is able to handle both DT_INT32 and DT_INT64,
+// no need to have TypeConstraint for `Tidx`
+#define REGISTER_KERNEL(type)                                           \
```

Comment:
Register a GPU kernel as well? That might require adding a .device() before the eigen call to .broadcast but otherwise should be easy.